Lib/test/test_concurrent_futures.py

```diff
@@ -315,10 +315,10 @@ def _prime_executor(self):
         pass
 
     def test_threads_terminate(self):
-        self.executor.submit(mul, 21, 2)
-        self.executor.submit(mul, 6, 7)
-        self.executor.submit(mul, 3, 14)
-        self.assertEqual(len(self.executor._threads), 3)
+        self.executor.submit(mul, 21, 2).result()
+        self.executor.submit(mul, 6, 7).result()
+        self.executor.submit(mul, 3, 14).result()
```

Comment:
I'd rather see the test fixed to test the proper condition (i.e. all 3 threads launched, then properly terminated).
For example (untested):
```
def acquire_lock(lock):
    lock.acquire()

def test_threads_terminate(self):
    sem = threading.Semaphore(0)
    for i in range(3):
        self.executor.submit(acquire_lock, sem)
    self.assertEqual(len(self.executor._threads), 3)
    for i in range(3):
        sem.release()
    # etc.
```